controllers/periodic_healthcheck.go

```diff
@@ -53,28 +52,34 @@ func (r *EtcdadmClusterReconciler) startHealthCheckLoop(ctx context.Context, don
 				continue
 			}
 			for _, ec := range etcdClusters.Items {
-				if !ec.Status.CreationComplete {
-					// etcdCluster not fully provisioned yet
+				if annotations.HasPaused(&ec) {
+					r.Log.Info("EtcdadmCluster reconciliation is paused, skipping health checks", "EtcdadmCluster", ec.Name)
 					continue
 				}
-				if annotations.HasPaused(&ec) {
-					r.Log.Info(fmt.Sprintf("Reconciliation is paused for etcdadmCluster %s", ec.Name))
+				if conditions.IsFalse(&ec, etcdv1.EtcdCertificatesAvailableCondition) {
+					r.Log.Info("EtcdadmCluster certificates are not ready, skipping health checks", "EtcdadmCluster", ec.Name)
```

Comment:
For adding objects to controller logs, this is super useful
```suggestion
					r.Log.Info("EtcdadmCluster certificates are not ready, skipping health checks", "cluster", klog.KObj(ec))	
```
No need to change the other logs, but you could do this at the top of the loop
```go
log = r.Log.WithValues("cluster", klog.KObj(ec))
```

and reuse it everywhere else